test/unit/browser/pages/vscode.test.ts

```diff
@@ -175,4 +177,151 @@ describe("vscode", () => {
       localStorage.removeItem("colorThemeData")
     })
   })
+  describe("getLoader", () => {
+    it("should throw an error if options.csStaticBase is undefined or an empty string in context", () => {
```

Comment:
My bad, I miscommunicated. This just tests that the test itself throws which I don't think we need to have.

Take for example `setBodyBackgroundToThemeBackgroundColor`. It claims to take a document but checks if it's undefined. But it can only be a document according to the types.

So there are two solutions to this mismatch:

1. Change the type of `document` on `setBodyBackgroundToThemeBackgroundColor` to `Document | undefined`. Now it truly reflects what the function expects to handle.
2. Leave the type. This means that code which calls the function needs to be sure it is calling it with something valid. Usually if the types say it's valid it is so nothing needs to be done. But if something does need to be checked then we'd have something on the calling side. In this example we'd have something like this inside `main`:

```
if (!document) throw new Error()
setBodyBackgroundToThemeBackgroundColor(document)
```

However in our case we can make `main` also take a `document` negating the need for any checks.

Reply:
Or another way of saying it is that manual typechecks are possibly best done at the boundaries of the code where the types first enter (idk if there's any formal proof or evidence of this though haha).

Reply:
> However in our case we can make main also take a document negating the need for any checks.

Actually, why don't I do that instead and have the error live there? We have multiple uses for `Document` so it would make sense to do that

Reply:
Makes sense! Maybe eventually we could have something like this in each top-level browser file:

```
try {
  assertExists(document, window) // We make sure the globals we expect exist
  main(document, window) // Then pass in the globals so all our functions are pure(ish)
} catch (error) {
  console.log(error)
}
```

And we could create `assertExists` in one of our util files. Just a thought though, doesn't have to be this way.

Reply:
Great idea! I like that approach